src/lib/Exporter/Exporter.php

```diff
@@ -218,7 +197,7 @@ private function generateFiles(array $languages, string $chunkDir, array $option
     {
         $urls = [];
 
-        $output->writeln(sprintf('Exporting %s content types', count($options['contentTypeIds'])));
+        $output->writeln(sprintf('Exporting %s content types', \count($options['contentTypeIds'])));
 
```

Comment:
I see, that using sudo solves the problem. Using a specific user would allow to control access in detail though.
In https://doc.ezplatform.com/en/latest/guide/personalization/#4-allow-public-https-access we already describe to create an eZ Platform User for basic auth:

> Create a User with the name of the customerID and a password which is the license key in your local security provider. 

We could use this user to check access to content :

```
        // check if user with login == $options['customerId'] exists
        $customerUser = null;
        try {
            $customerUser = $this->repository->getUserService()->loadUserByLogin((string)$options['customerId']);
        } catch (NotFoundException $e) {
            // void
        }

        // switch to $customerUser
        if ($customerUser !== null) {
            $this->repository->getPermissionResolver()->setCurrentUserReference(
                $customerUser
            );
        }
````

Reply:
Maybe, but then you'll need to configure it, for exporting it might be we should expect to export everything regardless of permissions? If so sudo, and then assuming this isn't exposed anywhere.

Reply:
We are already using an eZ Platform User for checking access when a single content is updated from backoffice.
Not doing the same for the full export might lead to a situation where a specific content is exported but updates from backoffice are not working.

Reply:
@reithor the article from the link provided by you states that the User is required when you want to limit access to the rest endpoint:
```
security:
    providers:
        ezpublish:
            id: ezpublish.security.user_provider
    firewalls:
        ezpublish_rest:
            pattern: ^/api/ezp/v2
            stateless: true
            ezpublish_http_basic:
                realm: eZ Publish REST API
```
So my best guess is only done to limit access to the endpoint, not to the content itself. Is that correct? If so, then the user is only used for authorization, not for Content permissions per se.

Reply:
Actually the doc is not correct here imho - I think we should sync in a meeting to exchange current status
Maybe together with @ciastektk ? I have a demo where i can show, what is actually working and what is not working.

Reply:
I had a discuss with backend team about using "sudo" and for purposes like this it is ok. But we need to use sudo in https://github.com/ezsystems/ezrecommendation-client/blob/v1.1/src/bundle/Controller/ContentController.php and https://github.com/ezsystems/ezrecommendation-client/blob/v1.1/src/bundle/Controller/ContentTypeController.php as well. 

I'd to know what exactly not working so sync in a meeting is a good idea. Today I have time after 2 p.m.

Reply:
@ciastektk & @mateuszbieniek  - a sum up of our call:

Using sudo in Exporter + doing the same in ContentController and ContentTypeController is the way to go for now.
This will guarantee that all content of configured content types will be synced to recommendation service.
This applies to full export (using Exporter) and "incremental" export (when Content is updated via Admin UI).

More granalur options limiting access to content via roles/policies are doable - but this needs a higher level discussion regarding use cases/needs and amount of work.

Reply:
Pinging @andrerom and @reithor for another round of CR, so we can push it forward :)